src/jaxsim/rbda/utils.py

```diff
@@ -131,6 +132,13 @@ def process_inputs(
     if W_Q_B.shape != (4,):
         raise ValueError(W_Q_B.shape, (4,))
 
+    # Check that the quaternion is unary since our RBDAs make this assumption in order
+    # to prevent introducing additional normalizations that would affect AD.
+    exceptions.raise_value_error_if(
+        condition=jnp.logical_not(jnp.allclose(W_Q_B.dot(W_Q_B), 1.0)),
+        msg="A RBDA received a quaternion that is not normalized.",
+    )
```

Comment:
You can check an example where a similar exception is raised in https://github.com/ami-iit/jaxsim/actions/runs/9895886879/job/27336807249#step:11:606. As you could see, there is already the printed (full) stack that provides all the needed information. Do you think it can still be valuable adding the frame in the exception message?

```
/opt/hostedtoolcache/Python/3.10.14/x64/lib/python3.10/site-packages/jax/_src/public_test_util.py:309: in check_grads
    _check_grads(f, args, order)
/opt/hostedtoolcache/Python/3.10.14/x64/lib/python3.10/site-packages/jax/_src/public_test_util.py:296: in _check_grads
    _check_jvp(f, partial(api.jvp, f), args, err_msg=fwd_msg)
/opt/hostedtoolcache/Python/3.10.14/x64/lib/python3.10/site-packages/jax/_src/public_test_util.py:238: in check_jvp
    t_out_expected = numerical_jvp(f, args, tangent, eps=eps)
/opt/hostedtoolcache/Python/3.10.14/x64/lib/python3.10/site-packages/jax/_src/public_test_util.py:213: in numerical_jvp
    f_pos = f(*add(primals, delta))
tests/test_automatic_differentiation.py:93: in <lambda>
    aba = lambda W_p_B, W_Q_B, s, W_v_WB, ṡ, τ, W_f_L, g: jaxsim.rbda.aba(
/opt/hostedtoolcache/Python/3.10.14/x64/lib/python3.10/site-packages/jaxsim/rbda/aba.py:52: in aba
    W_p_B, W_Q_B, s, W_v_WB, ṡ, _, _, τ, W_f, W_g = utils.process_inputs(
/opt/hostedtoolcache/Python/3.10.14/x64/lib/python3.10/site-packages/jaxsim/rbda/utils.py:137: in process_inputs
    exceptions.raise_value_error_if(
/opt/hostedtoolcache/Python/3.10.14/x64/lib/python3.10/site-packages/jaxsim/exceptions.py:63: in raise_value_error_if
    return raise_if(condition, ValueError, msg, *args, **kwargs)
/opt/hostedtoolcache/Python/3.10.14/x64/lib/python3.10/site-packages/jaxsim/exceptions.py:49: in raise_if
    return _run_callback_only_if_condition_is_true(*args, **kwargs)
```

Reply:
If we have already the full stack, then it's already good! Thanks for the explanation